store/modules/mission.js

```diff
@@ -31,9 +34,21 @@ const mutations = {
   [MISSION_SET_MISSION_CLAIMED](state, missionId) {
     state.missions = state.missions.filter(m => m.id !== missionId);
   },
+  [MISSION_STEP_MISSION](state, { missionId, taskId }) {
+    const mission = state.missions.find(m => m.id === missionId);
+    Vue.set(mission, taskId, true);
+  },
+  [MISSION_SET_MISSION_DONE](state, { missionId }) {
+    const mission = state.missions.find(m => m.id === missionId);
+    Vue.set(mission, 'done', true);
+  },
   [MISSION_SET_REFERRAL_LIST](state, referrals) {
     state.referrals = referrals;
   },
+  [MISSION_SET_REFERRAL_AVATAR](state, { userId, avatar }) {
+    const referral = state.referrals.find(r => r.id === userId);
```

Comment:
Same as above for `MISSION_STEP_MISSION`, `MISSION_SET_MISSION_DONE`, `MISSION_SET_REFERRAL_AVATAR`